src/entry-client.js

```diff
@@ -130,12 +130,18 @@ export default function initializeClient(createApp, clientOpts) {
                     .filter(c => !shouldIgnoreRouteUpdate(c, fetchDataArgs))
                     .forEach((c) => {
                         const name = getModuleName(c, to);
-                        const existingModule = find(registeredModules, { name });
+                        const existingModule = get(store, `_modulesNamespaceMap.${name}/`);
                         if (existingModule) {
-                            // We already have this module registered, update the
-                            // index to mark it as recent
                             opts.logger.info('Skipping duplicate Vuex module registration:', name);
-                            existingModule.index = moduleIndex++;
+                            // If the module was registered outside of the routing flow,
+                            // we need to add it to registeredModules.
+                            // Otherwise we will update the index to mark it as recent
+                            const registeredModule = find(registeredModules, { name });
+                            if (!registeredModule) {
+                                registeredModules.push({ name, index: moduleIndex++ });
+                            } else {
+                                registeredModule.index = moduleIndex++;
+                            }
```

Comment:
yup, this is exactly what I was thinking when I was debugging. The module is already registered. So no need to actually re-register it when the view loads. Just need to recognize it's there.